### PR TITLE
t-hmi: fix sdcard class for 1bit and add touchhandler class

### PR DESCRIPTION
**Pull request by cosmicpsyop**
- SDCard enable; two separate issues
    - proper detecting of the sdcard by reducing the SD_MMC frequency
details at https://github.com/Xinyuan-LilyGO/T-HMI/issues/27#issuecomment-2132046978
    - class SDcard blows up in 1bit mode requiring `initSDCard()` move to public
- Touch screen basic support 
    - Top 3/4 advances through screens
    - Bottom 1/4 switch lower between workers and fees
- Remove vkbit pool from mem worker API discovery because [PRl #410](https://github.com/BitMaker-hub/NerdMiner_v2/pull/410)
- Update OneButton to 2.5.0 

See [Video>>](https://youtu.be/El1wv85a0DI)

### CODE REVIEW

File: src/drivers/storage/SDCard.h [hunk 1]
```diff
@@ -58,8 +58,13 @@ class SDCard
     bool loadConfigFile(TSettings* Settings);
     bool cardAvailable();
     bool cardBusy();
+#ifdef NERDMINER_T_HMI
```
Comment (julio-matarranz) [inline src/drivers/storage/SDCard.h:61]: Is this code only related to T-HMI or is there a more generic name for this condition ?

Comment (cosmicpsyop) [reply src/drivers/storage/SDCard.h:61]: just being careful not to eff up other builds and cause a regression.  it would have been easier to make that method public in general.

File: src/NerdMinerV2.ino.cpp [hunk 2]
```diff
@@ -14,6 +14,7 @@
 #include "drivers/displays/display.h"
 #include "drivers/storage/SDCard.h"
 #include "timeconst.h"
+#include "TouchHandler.h"
```
Comment (julio-matarranz) [inline src/NerdMinerV2.ino.cpp:17]: No condition in that line ?

File: src/NerdMinerV2.ino.cpp [hunk 3]
```diff
@@ -96,7 +101,11 @@ void setup()
   /******** SHOW LED INIT STATUS (devices without screen) *****/
   mMonitor.NerdStatus = NM_waitingConfig;
   doLedStuff(0);
-  
+
+#ifdef NERDMINER_T_HMI
```
Comment (julio-matarranz) [inline src/NerdMinerV2.ino.cpp:105]: Here too, could the condition can be another thing more generic like  "TOUCH_ENABLE" ?

Comment (cosmicpsyop) [reply src/NerdMinerV2.ino.cpp:105]: yeah,  good one.

File: src/NerdMinerV2.ino.cpp (hunk 2 above)
Comment (cosmicpsyop) [reply src/NerdMinerV2.ino.cpp:19]: got it.

File: src/drivers/storage/SDCard.cpp [hunk 4]
```diff
@@ -167,7 +169,12 @@ bool SDCard::initSDcard()
 #elif defined (BUILD_SDMMC_1)
 #warning SDMMC : 1 - bit mode is not always working. If you experience issues, try other modes.
         iSD_->setPins(SDMMC_CLK, SDMMC_CMD, SDMMC_D0);
+#ifdef NERDMINER_T_HMI
```
Comment (julio-matarranz) [inline src/drivers/storage/SDCard.cpp:172]: you could define a SD_FREQUENCY variable
so it could be tunable for every board

File: src/monitor.cpp [hunk 5]
```diff
@@ -335,13 +335,12 @@ String getPoolAPIUrl(void) {
         else {
             switch (Settings.PoolPort) {
                 case 3333:
-                    if (Settings.PoolAddress == "pool.vkbit.com")
-                        poolAPIUrl = "https://vkbit.com/miner/";
-                    else if (Settings.PoolAddress == "pool.sethforprivacy.com")
+                    if (Settings.PoolAddress == "pool.sethforprivacy.com")
```
Comment (julio-matarranz) [inline src/monitor.cpp:338]: this change doesn't seem to be related with touchhandler and 1bit fix

Comment (cosmicpsyop) [reply src/monitor.cpp:338]: yup. hijacked pr to correct my unreleased code after reading https://github.com/BitMaker-hub/NerdMiner_v2/pull/410  not clear on the reasoning,  it did  function at the time of testing.
This adjustment is mentioned in this pr summary along with OneButton.

File: src/drivers/storage/SDCard.cpp (hunk 4 above)
Comment (cosmicpsyop) [reply src/drivers/storage/SDCard.cpp:172]: A '#define'  seems reasonable here
```
#define SDMMC_FREQ_20MHZ  20000
#define SDMMC_FREQ_10MHZ  10000
#define SDMMC_FREQ_5MHZ   5000
#define SDMMC_FREQ_2MHZ   2000
```
but a class refactor of some sort would be needed if I understand the suggestion.
```
class SDCard {
public:
    SDCard(uint32_t frequency = SDMMC_FREQ_20MHZ);
    void setFrequency(uint32_t frequency);

private:
    uint32_t frequency;
};
```
https://github.com/espressif/esp-idf/blob/0453e8608bde98133a427a74ae61d272770b1bfd/components/sdmmc/include/sd_protocol_types.h#L187

Comment (julio-matarranz) [reply src/drivers/storage/SDCard.cpp:172]: > A '#define' seems reasonable here
> 
> ```
> #define SDMMC_FREQ_20MHZ  20000
> #define SDMMC_FREQ_10MHZ  10000
> #define SDMMC_FREQ_5MHZ   5000
> #define SDMMC_FREQ_2MHZ   2000
> ```
> 
> but a class refactor of some sort would be needed if I understand the suggestion.
> 
> ```
> class SDCard {
> public:
>     SDCard(uint32_t frequency = SDMMC_FREQ_20MHZ);
>     void setFrequency(uint32_t frequency);
> 
> private:
>     uint32_t frequency;
> };
> ```
> 
> https://github.com/espressif/esp-idf/blob/0453e8608bde98133a427a74ae61d272770b1bfd/components/sdmmc/include/sd_protocol_types.h#L187

Dear,

#ifdef SD_FREQUENCY
        cardInitialized_ = iSD_->begin("/sd", true, false, SD_FREQUENCY);
#else
        cardInitialized_ = iSD_->begin("/sd", true);
#endif

and you can add -D SD_FREQUENCY=20000 inside platformio.ini or #define SD_FREQUENCY 20000 inside lilygoT_HMI.h

Comment (cosmicpsyop) [reply src/drivers/storage/SDCard.cpp]: simplicity is elegant
